### ISSUE TITLE
Validate Vertex type for Serialiser.

### ISSUE THREAD

**Issue by GCHQDeveloper404** [gchq/Gaffer#1062]
If a graph is defined without the vertex type there will be a null pointer when the vertex is serialised.

```
new Graph.Builder()
                .addSchema(new Schema.Builder()
                        .entity("testGroup")
                        .build())
                .storeProperties(StreamUtil.STORE_PROPERTIES)
                .build();
```

or remove it?  `.entity("testGroup")`

**Comment by p013570** [comment]
Same applies to .edge("testGroup")

**Comment by m607123** [comment]
Taken over as of 26th July

**Comment by m607123** [comment]
# Migration

When creating a schema, the source, destination, and vertex fields are now required to pass validation.

**Comment by p013570** [comment]
Merged into develop-0.8.0
